src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -136,10 +155,40 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
   closePopover() {
     this.setState({ ...this.state, isPopoverOpen: false });
   }
+  /**
+   * when call handleSwitchDefaultDatasource, the default data source must exist,
+   * since the button only display when defaultDataSource exist
+   */
+  async handleSwitchDefaultDatasource(defaultDataSourceId: string | null) {
+    /**
+     * default dataSource label must exist when defaultDataSource exist
+     */
+    const defaultDataSourceLabel = (
+      await getDataSourceById(defaultDataSourceId!, this.props.savedObjectsClient!)
```

Comment:
What if this getDataSourceById failed?

Reply:
nice catch, while basically, we assume the defaultDataSource is a valid data source object, we can always get it. If exception happens, we are already in `error` state, the popover and toast will keep open until we manually close it or the switch work with valid defaultDataSource option.

Reply:
Okay. So if there is a internet connection issue, there still is a possibilities that getDataSourcById would fail. Maybe we should have this getDataSourcById  in the componentDidMount try block to catch the error if it fails. Not a blocking issue.

Reply:
I thinking wrapping another api call in catch block will be very, we can either set the default data source label when we set it in the uiSettings, or, call getDataSourceById in the first try to get it label, then we can handle either error when getDataSource for the activeOption or defaultDataSource, to have this, I  also recommend maintaining another state call defaultDataSourceOption as DataSourceOption `{id: defaultDataSourceId, label: getDataSourceById(defaultDataSourceId).title}`

Reply:
For refactoring uiSettings to have defaultDataSource as `DataSourceOption`, I think better to have a separate issue to track, and we need to make sure all the other components' behavior will not be impacted by this change. 
The benefit of this change can add the request call for defaultDataSource out of any dataSource component since it is in uiSettings
While at another side, not all component/plugin use defaultDataSource, and not all of them will need to use default data source as the data source option